algebra/localization.lean

```diff
@@ -86,6 +97,169 @@ by refine
   apply quotient.sound,
   existsi (1:α),
   existsi is_submonoid.one_mem S,
-  ring }
+  simp [mul_left_comm, mul_add, mul_comm] }
+
+def of_comm_ring : α → loc α S :=
+λ r, ⟦⟨r, 1, is_submonoid.one_mem S⟩⟧
+
+instance : is_hom (of_comm_ring α S) :=
+{ map_add := λ x y, quotient.sound $ by simp,
+  map_mul := λ x y, quotient.sound $ by simp,
+  map_one := rfl }
+
+local infix ^ := monoid.pow
+
+variable {α}
+
+def powers (x : α) : set α := {y | ∃ n, x^n = y}
+
+instance powers.is_submonoid (x : α) : is_submonoid α (powers x) :=
+{ one_mem := ⟨0, by simp⟩,
+  mul_mem := λ x₁ x₂ ⟨n₁, hn₁⟩ ⟨n₂, hn₂⟩, ⟨n₁ + n₂, by simp [pow_add, *]⟩ }
+
+def away (x : α) := loc α (powers x)
```

Comment:
You will want to prove instances of comm_ring for these, so that you don't need to specify them manually like in `at_prime.local_ring`.